engine/src/main/java/org/terasology/rendering/shader/ShaderParametersBase.java

```diff
@@ -43,27 +45,29 @@ public void applyParameters(Material program) {
         program.setFloat("viewingDistance", CoreRegistry.get(Config.class).getRendering().getViewDistance().getChunkDistance().x * 8.0f);
 
         WorldRenderer worldRenderer = CoreRegistry.get(WorldRenderer.class);
-        WorldProvider worldProvider = CoreRegistry.get(WorldProvider.class);
+        BackdropProvider backdropProvider = CoreRegistry.get(BackdropProvider.class);
 
-        if (worldRenderer != null) {
-            program.setFloat("daylight", worldRenderer.getDaylight(), true);
+        if (worldRenderer != null && backdropProvider != null) {
```

Comment:
About  logic in the ShaderParameter classes: my bad. My sunday-evening-programmer standards are pretty low: anything more branched or more complex than straight variable assignments is a lot of logic. =D 

About everything else: I agree with the direction you are setting here. I'm still very fuzzy about the whole vision of it, but it sounds like a good vision. Should I cut&paste your last comment and edit it a little into an issue? Or is this something you'll be tackling within the gestalt work anyway?